pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -526,6 +526,48 @@ type PrometheusSpec struct {
 	// Defines the runtime reloadable configuration of the timeseries database
 	// (TSDB).
 	TSDB TSDBSpec `json:"tsdb,omitempty"`
+	// TracingConfig configures tracing in Prometheus. This is an experimental feature, it may change in any upcoming release in a breaking way.
+	// +optional
+	TracingConfig *PrometheusTracingConfig `json:"tracingConfig,omitempty"`
+}
+
+type PrometheusTracingConfig struct {
+	// Client used to export the traces. Options are "http" or "grpc".
+	//+kubebuilder:validation:Enum=http;grpc
+	//+kubebuilder:default:="grpc"
+	// +optional
+	ClientType *string `json:"clientType"`
+
+	// Endpoint to send the traces to. Should be provided in format <host>:<port>.
+	Endpoint string `json:"endpoint"`
+
+	// Sets the probability a given trace will be sampled. Must be a float from 0 through 1.
+	//+kubebuilder:default:="0"
+	// +optional
+	SamplingFraction *string `json:"samplingFraction"`
```

Comment:
Maybe it was discussed before but representing float64 in CRDs is "challenging" (https://github.com/kubernetes-sigs/controller-tools/issues/245). Apparently one option could be to use `resource.Quantity` which should bring API validation but I haven't looked deeper into it.

Reply:
@simonpasquier I tried to use Quantity but for some reason (maybe I'm not getting right the API), if I set 0.5 when get the value it returns 0 🥲